doc/guides/arch/arm.rst

```diff
@@ -0,0 +1,79 @@
+| Single-thread kernel support    |                                   |   Y    |    Y   |   Y     |   Y    |    Y      |    Y   |    Y    |     Y      |   Y        |
++---------------------------------+-----------------------------------+--------+--------+---------+--------+-----------+--------+---------+------------+------------+
+| Thread local storage support    |                                   |   Y    |    Y   |   Y     |   Y    |    Y      |    Y   |    Y    |     Y      |   Y        |
++---------------------------------+-----------------------------------+--------+--------+---------+--------+-----------+--------+---------+------------+------------+
+| Interrupt handling              |                                   |                                                                                             |
++---------------------------------+-----------------------------------+--------+--------+---------+--------+-----------+--------+---------+------------+------------+
+|                                 |   Regular interrupts              |   Y    |    Y   |   Y     |   Y    |    Y      |    Y   |    Y    |     Y      |   Y        |
++---------------------------------+-----------------------------------+--------+--------+---------+--------+-----------+--------+---------+------------+------------+
+|                                 |   Dynamic interrupts              |   Y    |    Y   |   Y     |   Y    |    Y      |    Y   |    Y    |     Y      |   Y        |
++---------------------------------+-----------------------------------+--------+--------+---------+--------+-----------+--------+---------+------------+------------+
+|                                 |   Direct  interrupts              |   Y    |    Y   |   Y     |   Y    |    Y      |    Y   |    Y    |     Y      |   Y        |
++---------------------------------+-----------------------------------+--------+--------+---------+--------+-----------+--------+---------+------------+------------+
+|                                 |   Zero Latency interrupts         |   N    |    N   |   N     |   Y    |    Y      |    Y   |    Y    |     Y      |   Y        |
++---------------------------------+-----------------------------------+--------+--------+---------+--------+-----------+--------+---------+------------+------------+
+| CPU idling                      |                                   |   Y    |    Y   |   Y     |   Y    |    Y      |    Y   |    Y    |     Y      |   Y        |
++---------------------------------+-----------------------------------+--------+--------+---------+--------+-----------+--------+---------+------------+------------+
+| Native system timer (SysTick)   |                                   |   N    |    N   |   Y     |   Y    |    Y      |    Y   |    Y    |     Y      |   Y        |
```

Comment:
The Cortex-M1 IP has optional SysTick support. This is supported by Zephyr. Perhaps we need to expand the `Y`/`N` with `O` for optional?

Reply:
I was thinking about it, yes.
I think we need either some increased legend or some list of footnotes..

Reply:
Done, added footnote @henrikbrixandersen 
(this is how the ARC guys have populated their arch-support-status table: using foot notes :) )